# 1. Collect a GeoSurfaceTimePoint

In [6]:
"""
lat = -0.625
lon = -0.9375
time = "2017-07-15T06:00:00.000"

# filter
gstpFilter = c3.Filter().eq("latitude", lat).and_().eq("longitude", lon).and_().eq("time", time)

# lt or gt instead of eq for intervals

# collect instances 
gstp = c3.GeoSurfaceTimePoint.fetch({"filter": gstpFilter})
"""

lat1 = 0.0
lat2 = 10.0
lon1 = 0.0
lon2 = 10.0
time1 = "2017-07-01T00:00:00"
time2 = "2017-07-01T23:59:59"
gstpFilter = c3.Filter() \
    .ge("latitude", lat1) \
    .and_().le("latitude", lat2) \
    .and_().ge("longitude", lon1) \
    .and_().le("longitude", lon2) \
    .and_().ge("time", time1) \
    .and_().le("time", time2)
gstp = c3.GeoSurfaceTimePoint.fetch({"filter": gstpFilter})

In [7]:
gstp

c3.FetchResult<GeoSurfaceTimePoint>(
 objs=c3.Arry<GeoSurfaceTimePoint>([c3.GeoSurfaceTimePoint(
         id='0.625_0.938_2017-07-01T00:20:00',
         meta=c3.Meta(
                tenantTagId=151,
                tenant='dev',
                tag='tc02d',
                created=datetime.datetime(2022, 6, 2, 18, 0, 38, tzinfo=datetime.timezone.utc),
                createdBy='babreu@illinois.edu',
                updated=datetime.datetime(2022, 6, 2, 18, 0, 38, tzinfo=datetime.timezone.utc),
                updatedBy='babreu@illinois.edu',
                timestamp=datetime.datetime(2022, 6, 2, 18, 0, 38, tzinfo=datetime.timezone.utc),
                fetchInclude='[]',
                fetchType='GeoSurfaceTimePoint'),
         version=1,
         latitude=0.625,
         longitude=0.9375,
         time=datetime.datetime(2017, 7, 1, 0, 20)),
        c3.GeoSurfaceTimePoint(
         id='0.625_0.938_2017-07-01T03:20:00',
         meta=c3.Meta(
                tenantTagId=151,
        

Good, we found it! Now we get the object itself to use it as a key.

In [8]:
gstp = gstp.objs[0]

# 2. Use collected object as key to Simulation3HourlyAODOutput

In [9]:
import pandas as pd

In [10]:
# filter
s3haodFilter = c3.Filter().eq("geoSurfaceTimePoint", gstp) 

# collect objects
data = c3.Simulation3HourlyAODOutput.fetch({"filter": s3haodFilter})

# cast it into pandas
data = data.objs.toJson()
data = pd.DataFrame(data)

In [11]:
data

type                                       id  \
0    Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DRI45YJ   
1    Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DSTZASE   
2    Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DO4JCGZ   
3    Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DOWCISB   
4    Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DWGAIUR   
..                          ...                                      ...   
216  Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DP4SDFB   
217  Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DSZ6OWJ   
218  Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DXRICNR   
219  Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DRK5XSA   
220  Simulation3HourlyAODOutput  0.625_0.938_2017-07-01T00:20:00#DXMEJTM   

     version                                   simulationSample      dust  \
0          1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.003082   
1          1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.002954   
2          1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.002796   
3          1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.004310   
4          1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.001752   
..       ...                                                ...       ...   
216        1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.002958   
217        1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.002566   
218        1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.004754   
219        1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.002696   
220        1  {'type': 'SimulationSample', 'id': 'EnsNo_1_Si...  0.002948   

     solubleAitkenMode  solubleAccumulationMode  solubleCoarseMode  \
0             0.030086                 0.217025           0.018044   
1             0.012938                 0.243141           0.026136   
2             0.006634                 0.308470           0.080707   
3             0.026322                 0.184432           0.020594   
4             0.020261                 0.180450           0.028082   
..                 ...                      ...                ...   
216           0.018987                 0.281946           0.049134   
217           0.023673                 0.318031           0.012389   
218           0.005767                 0.192712           0.022504   
219           0.027969                 0.124522           0.007288   
220           0.006416                 0.252674           0.040027   

     insolubleAitkenMode                                geoSurfaceTimePoint  
0               0.007949  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  
1               0.012145  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  
2               0.025604  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  
3               0.013919  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  
4               0.014441  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  
..                   ...                                                ...  
216             0.009772  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  
217             0.018010  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  
218             0.005149  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  
219             0.006367  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  
220             0.022693  {'type': 'GeoSurfaceTimePoint', 'id': '0.625_0...  

[221 rows x 10 columns]

In [13]:
data.geoSurfaceTimePoint[0]

{'type': 'GeoSurfaceTimePoint', 'id': '0.625_0.938_2017-07-01T00:20:00'}

# 3. A weird observation?

We check how the fetched data over a time interval differ based on how the intervals are specified. Below we get around 
221 x 8 = 1768 rows of data for each day (which is what we expect) but only 2000 from both.

In [3]:
lat = -0.625
lon = -0.9375
start_time = "2017-07-15T00:00:00.000"
end_time = "2017-07-15T21:00:00.000"

# filter
gstpFilter = c3.Filter().eq("latitude", lat).and_().eq("longitude", lon).and_().ge("time", start_time).and_().le("time", end_time)

# lt or gt instead of eq for intervals

# collect instances 
gstp = c3.GeoSurfaceTimePoint.fetch({"filter": gstpFilter}).objs

# filter
s3haodFilter = c3.Filter().eq("geoSurfaceTimePoint", gstp) 

# collect objects
data = c3.Simulation3HourlyAODOutput.fetch({"filter": s3haodFilter})

# cast it into pandas
data = data.objs.toJson()
data = pd.DataFrame(data)

print(data.shape)

(1760, 10)


In [4]:
lat = -0.625
lon = -0.9375
start_time = "2017-07-16T00:00:00.000"
end_time = "2017-07-16T21:00:00.000"

# filter
gstpFilter = c3.Filter().eq("latitude", lat).and_().eq("longitude", lon).and_().ge("time", start_time).and_().le("time", end_time)

# lt or gt instead of eq for intervals

# collect instances 
gstp = c3.GeoSurfaceTimePoint.fetch({"filter": gstpFilter}).objs

# filter
s3haodFilter = c3.Filter().eq("geoSurfaceTimePoint", gstp) 

# collect objects
data = c3.Simulation3HourlyAODOutput.fetch({"filter": s3haodFilter})

# cast it into pandas
data = data.objs.toJson()
data = pd.DataFrame(data)

print(data.shape)

(1768, 10)


In [5]:
lat = -0.625
lon = -0.9375
start_time = "2017-07-15T00:00:00.000"
end_time = "2017-07-16T21:00:00.000"

# filter
gstpFilter = c3.Filter().eq("latitude", lat).and_().eq("longitude", lon).and_().ge("time", start_time).and_().le("time", end_time)

# lt or gt instead of eq for intervals

# collect instances 
gstp = c3.GeoSurfaceTimePoint.fetch({"filter": gstpFilter}).objs

# filter
s3haodFilter = c3.Filter().eq("geoSurfaceTimePoint", gstp) 

# collect objects
data = c3.Simulation3HourlyAODOutput.fetch({"filter": s3haodFilter})

# cast it into pandas
data = data.objs.toJson()
data = pd.DataFrame(data)

print(data.shape)

(2000, 10)
